akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaRange.scala

```diff
@@ -0,0 +1,163 @@
+    override def isMessage = mediaType.isMessage
+    override def isMultipart = mediaType.isMultipart
+    override def isText = mediaType.isText
+    override def isVideo = mediaType.isVideo
+    def matches(mediaType: MediaType) =
+      this.mediaType.mainType == mediaType.mainType && this.mediaType.subType == mediaType.subType
+    def withParams(params: Map[String, String]) = copy(mediaType = mediaType.withParams(params))
+    def withQValue(qValue: Float) = copy(qValue = qValue)
+    def render[R <: Rendering](r: R): r.type = if (qValue < 1.0f) r ~~ mediaType ~~ ";q=" ~~ qValue else r ~~ mediaType
+    def specimen = mediaType
+  }
+
+  implicit def apply(mediaType: MediaType): MediaRange = apply(mediaType, 1.0f)
+  def apply(mediaType: MediaType, qValue: Float = 1.0f): MediaRange = One(mediaType, qValue)
+}
+
+object MediaRanges extends ObjectRegistry[String, MediaRange] {
+
+  sealed abstract case class PredefinedMediaRange(value: String) extends MediaRange with LazyValueBytesRenderable {
+    val mainType = value takeWhile (_ != '/')
+    register(mainType, this)
+    def params = Map.empty
+    def qValue = 1.0f
+    def withParams(params: Map[String, String]) = MediaRange.custom(mainType, params)
+    def withQValue(qValue: Float) = if (qValue != 1.0f) MediaRange.custom(mainType, params, qValue) else this
+  }
+
+  val `*/*` = new PredefinedMediaRange("*/*") {
+    def matches(mediaType: MediaType) = true
+    def specimen = MediaTypes.`text/plain`
+  }
+  val `*/*;q=MIN` = `*/*`.withQValue(Float.MinPositiveValue)
+  val `application/*` = new PredefinedMediaRange("application/*") {
+    def matches(mediaType: MediaType) = mediaType.isApplication
+    override def isApplication = true
+    def specimen = MediaTypes.`application/json`
+  }
+  val `audio/*` = new PredefinedMediaRange("audio/*") {
+    def matches(mediaType: MediaType) = mediaType.isAudio
+    override def isAudio = true
+    def specimen = MediaTypes.`audio/ogg`
+  }
+  val `image/*` = new PredefinedMediaRange("image/*") {
+    def matches(mediaType: MediaType) = mediaType.isImage
+    override def isImage = true
+    def specimen = MediaTypes.`image/png`
+  }
+  val `message/*` = new PredefinedMediaRange("message/*") {
+    def matches(mediaType: MediaType) = mediaType.isMessage
+    override def isMessage = true
+    def specimen = MediaTypes.`message/rfc822`
+  }
+  val `multipart/*` = new PredefinedMediaRange("multipart/*") {
+    def matches(mediaType: MediaType) = mediaType.isMultipart
+    override def isMultipart = true
+    def specimen = MediaTypes.`multipart/form-data`
+  }
+  val `text/*` = new PredefinedMediaRange("text/*") {
+    def matches(mediaType: MediaType) = mediaType.isText
+    override def isText = true
+    def specimen = MediaTypes.`text/plain`
+  }
+  val `video/*` = new PredefinedMediaRange("video/*") {
+    def matches(mediaType: MediaType) = mediaType.isVideo
+    override def isVideo = true
+    def specimen = MediaTypes.`video/mp4`
```

Comment:
Yes, any will do.
Bascially, if someone says they accept `video/*` they should also accept `video/mp4`.
Very rarely used here, I'm sure, but required for completeness.